### PR TITLE
modified guess_atom_element for more accurate guess

### LINKED ISSUE #4167: Atom type guessing error when parse gro file
## Expected behavior ##

When I try to parse following gro file using `MDAnalysis.Universe`, 
<details>
  <summary>Example gro file</summary>

> mCP
 &nbsp;52
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C00 &nbsp; &nbsp; &nbsp; &nbsp;1 &nbsp; &nbsp; &nbsp;1.382 &nbsp; &nbsp; &nbsp;1.369 &nbsp; &nbsp; &nbsp;3.583 &nbsp; &nbsp;0.0621 &nbsp;-0.1579 &nbsp;-0.3177
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C01 &nbsp; &nbsp; &nbsp; &nbsp;2 &nbsp; &nbsp; &nbsp;1.516 &nbsp; &nbsp; &nbsp;1.408 &nbsp; &nbsp; &nbsp;3.551 &nbsp; &nbsp;0.2861 &nbsp; &nbsp;0.3115 &nbsp;-0.4733
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C02 &nbsp; &nbsp; &nbsp; &nbsp;3 &nbsp; &nbsp; &nbsp;1.335 &nbsp; &nbsp; &nbsp;1.406 &nbsp; &nbsp; &nbsp;3.711 &nbsp;-0.1396 &nbsp; &nbsp;0.1043 &nbsp;-0.6418
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C03 &nbsp; &nbsp; &nbsp; &nbsp;4 &nbsp; &nbsp; &nbsp;1.413 &nbsp; &nbsp; &nbsp;1.470 &nbsp; &nbsp; &nbsp;3.804 &nbsp; &nbsp;0.2561 &nbsp; &nbsp;0.2800 &nbsp; &nbsp;0.2722
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C04 &nbsp; &nbsp; &nbsp; &nbsp;5 &nbsp; &nbsp; &nbsp;1.603 &nbsp; &nbsp; &nbsp;1.465 &nbsp; &nbsp; &nbsp;3.646 &nbsp;-0.9943 &nbsp; &nbsp;0.1843 &nbsp;-0.0949
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C05 &nbsp; &nbsp; &nbsp; &nbsp;6 &nbsp; &nbsp; &nbsp;1.547 &nbsp; &nbsp; &nbsp;1.498 &nbsp; &nbsp; &nbsp;3.773 &nbsp; &nbsp;0.0720 &nbsp; &nbsp;0.3654 &nbsp; &nbsp;0.5643
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H06 &nbsp; &nbsp; &nbsp; &nbsp;7 &nbsp; &nbsp; &nbsp;1.705 &nbsp; &nbsp; &nbsp;1.487 &nbsp; &nbsp; &nbsp;3.617 &nbsp;-2.1881 &nbsp;-1.6600 &nbsp; &nbsp;1.6244
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H07 &nbsp; &nbsp; &nbsp; &nbsp;8 &nbsp; &nbsp; &nbsp;1.326 &nbsp; &nbsp; &nbsp;1.312 &nbsp; &nbsp; &nbsp;3.512 &nbsp;-0.1227 &nbsp; &nbsp;1.6765 &nbsp; &nbsp;3.4071
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H08 &nbsp; &nbsp; &nbsp; &nbsp;9 &nbsp; &nbsp; &nbsp;1.373 &nbsp; &nbsp; &nbsp;1.505 &nbsp; &nbsp; &nbsp;3.904 &nbsp;-2.8380 &nbsp; &nbsp;0.2915 &nbsp; &nbsp;0.0254
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H09 &nbsp; &nbsp; &nbsp;10 &nbsp; &nbsp; &nbsp;1.609 &nbsp; &nbsp; &nbsp;1.543 &nbsp; &nbsp; &nbsp;3.847 &nbsp;-0.9918 &nbsp; &nbsp;0.6125 &nbsp; &nbsp;0.1841
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;N0A &nbsp; &nbsp; &nbsp;11 &nbsp; &nbsp; &nbsp;1.569 &nbsp; &nbsp; &nbsp;1.360 &nbsp; &nbsp; &nbsp;3.423 &nbsp; &nbsp;0.7252 &nbsp; &nbsp;0.4807 &nbsp; &nbsp;0.1006
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;N0B &nbsp; &nbsp; &nbsp;12 &nbsp; &nbsp; &nbsp;1.199 &nbsp; &nbsp; &nbsp;1.378 &nbsp; &nbsp; &nbsp;3.753 &nbsp; &nbsp;0.0195 &nbsp; &nbsp;0.6945 &nbsp;-0.3455
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0C &nbsp; &nbsp; &nbsp;13 &nbsp; &nbsp; &nbsp;1.511 &nbsp; &nbsp; &nbsp;1.382 &nbsp; &nbsp; &nbsp;3.298 &nbsp;-0.0718 &nbsp; &nbsp;0.0881 &nbsp; &nbsp;0.4777
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0D &nbsp; &nbsp; &nbsp;14 &nbsp; &nbsp; &nbsp;1.693 &nbsp; &nbsp; &nbsp;1.291 &nbsp; &nbsp; &nbsp;3.403 &nbsp;-0.8847 &nbsp; &nbsp;0.5669 &nbsp;-0.3861
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0E &nbsp; &nbsp; &nbsp;15 &nbsp; &nbsp; &nbsp;1.079 &nbsp; &nbsp; &nbsp;1.390 &nbsp; &nbsp; &nbsp;3.677 &nbsp; &nbsp;0.1428 &nbsp;-0.3040 &nbsp;-0.4933
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0F &nbsp; &nbsp; &nbsp;16 &nbsp; &nbsp; &nbsp;1.157 &nbsp; &nbsp; &nbsp;1.323 &nbsp; &nbsp; &nbsp;3.880 &nbsp;-1.2695 &nbsp; &nbsp;0.3701 &nbsp; &nbsp;0.7727
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0G &nbsp; &nbsp; &nbsp;17 &nbsp; &nbsp; &nbsp;0.967 &nbsp; &nbsp; &nbsp;1.354 &nbsp; &nbsp; &nbsp;3.754 &nbsp; &nbsp;0.1120 &nbsp;-0.1261 &nbsp; &nbsp;0.2826
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0H &nbsp; &nbsp; &nbsp;18 &nbsp; &nbsp; &nbsp;1.019 &nbsp; &nbsp; &nbsp;1.302 &nbsp; &nbsp; &nbsp;3.872 &nbsp; &nbsp;0.6453 &nbsp; &nbsp;0.4619 &nbsp;-0.6466
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0I &nbsp; &nbsp; &nbsp;19 &nbsp; &nbsp; &nbsp;1.706 &nbsp; &nbsp; &nbsp;1.270 &nbsp; &nbsp; &nbsp;3.258 &nbsp;-0.1344 &nbsp; &nbsp;0.4492 &nbsp; &nbsp;0.3150
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0J &nbsp; &nbsp; &nbsp;20 &nbsp; &nbsp; &nbsp;1.595 &nbsp; &nbsp; &nbsp;1.327 &nbsp; &nbsp; &nbsp;3.198 &nbsp;-0.3651 &nbsp; &nbsp;0.3864 &nbsp; &nbsp;0.9254
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0K &nbsp; &nbsp; &nbsp;21 &nbsp; &nbsp; &nbsp;1.563 &nbsp; &nbsp; &nbsp;1.348 &nbsp; &nbsp; &nbsp;3.060 &nbsp; &nbsp;0.0239 &nbsp; &nbsp;0.2715 &nbsp; &nbsp;1.5069
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0M &nbsp; &nbsp; &nbsp;22 &nbsp; &nbsp; &nbsp;1.395 &nbsp; &nbsp; &nbsp;1.451 &nbsp; &nbsp; &nbsp;3.278 &nbsp; &nbsp;0.3359 &nbsp; &nbsp;0.8552 &nbsp; &nbsp;0.0206
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0N &nbsp; &nbsp; &nbsp;23 &nbsp; &nbsp; &nbsp;1.820 &nbsp; &nbsp; &nbsp;1.205 &nbsp; &nbsp; &nbsp;3.211 &nbsp;-0.8869 &nbsp; &nbsp;0.1811 &nbsp; &nbsp;0.0699
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0O &nbsp; &nbsp; &nbsp;24 &nbsp; &nbsp; &nbsp;1.784 &nbsp; &nbsp; &nbsp;1.254 &nbsp; &nbsp; &nbsp;3.501 &nbsp; &nbsp;0.2992 &nbsp;-0.1149 &nbsp;-0.4235
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0P &nbsp; &nbsp; &nbsp;25 &nbsp; &nbsp; &nbsp;1.084 &nbsp; &nbsp; &nbsp;1.464 &nbsp; &nbsp; &nbsp;3.558 &nbsp; &nbsp;0.0214 &nbsp;-0.0379 &nbsp; &nbsp;0.1615
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0Q &nbsp; &nbsp; &nbsp;26 &nbsp; &nbsp; &nbsp;0.839 &nbsp; &nbsp; &nbsp;1.378 &nbsp; &nbsp; &nbsp;3.699 &nbsp;-0.0488 &nbsp; &nbsp;0.1386 &nbsp;-0.0709
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0R &nbsp; &nbsp; &nbsp;27 &nbsp; &nbsp; &nbsp;1.239 &nbsp; &nbsp; &nbsp;1.285 &nbsp; &nbsp; &nbsp;3.989 &nbsp; &nbsp;0.3367 &nbsp;-0.1765 &nbsp;-0.0590
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0S &nbsp; &nbsp; &nbsp;28 &nbsp; &nbsp; &nbsp;0.957 &nbsp; &nbsp; &nbsp;1.237 &nbsp; &nbsp; &nbsp;3.982 &nbsp; &nbsp;0.1652 &nbsp; &nbsp;0.3390 &nbsp;-0.8689
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0T &nbsp; &nbsp; &nbsp;29 &nbsp; &nbsp; &nbsp;1.175 &nbsp; &nbsp; &nbsp;1.229 &nbsp; &nbsp; &nbsp;0.053 &nbsp;-0.1744 &nbsp; &nbsp;0.0147 &nbsp;-0.0595
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0U &nbsp; &nbsp; &nbsp;30 &nbsp; &nbsp; &nbsp;1.035 &nbsp; &nbsp; &nbsp;1.198 &nbsp; &nbsp; &nbsp;0.045 &nbsp;-0.0222 &nbsp; &nbsp;0.4244 &nbsp;-0.4755
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0V &nbsp; &nbsp; &nbsp;31 &nbsp; &nbsp; &nbsp;0.951 &nbsp; &nbsp; &nbsp;1.491 &nbsp; &nbsp; &nbsp;3.510 &nbsp; &nbsp;0.2775 &nbsp;-0.1529 &nbsp; &nbsp;0.3318
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0W &nbsp; &nbsp; &nbsp;32 &nbsp; &nbsp; &nbsp;0.832 &nbsp; &nbsp; &nbsp;1.442 &nbsp; &nbsp; &nbsp;3.573 &nbsp; &nbsp;0.1285 &nbsp;-0.4091 &nbsp;-0.5051
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0X &nbsp; &nbsp; &nbsp;33 &nbsp; &nbsp; &nbsp;1.440 &nbsp; &nbsp; &nbsp;1.413 &nbsp; &nbsp; &nbsp;3.035 &nbsp; &nbsp;0.0008 &nbsp; &nbsp;0.2347 &nbsp;-0.2036
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0Y &nbsp; &nbsp; &nbsp;34 &nbsp; &nbsp; &nbsp;1.357 &nbsp; &nbsp; &nbsp;1.470 &nbsp; &nbsp; &nbsp;3.141 &nbsp;-0.4632 &nbsp; &nbsp;1.0063 &nbsp;-0.2381
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C0Z &nbsp; &nbsp; &nbsp;35 &nbsp; &nbsp; &nbsp;1.913 &nbsp; &nbsp; &nbsp;1.162 &nbsp; &nbsp; &nbsp;3.309 &nbsp;-0.7378 &nbsp; &nbsp;0.4248 &nbsp;-0.2854
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;C10 &nbsp; &nbsp; &nbsp;36 &nbsp; &nbsp; &nbsp;1.894 &nbsp; &nbsp; &nbsp;1.177 &nbsp; &nbsp; &nbsp;3.450 &nbsp; &nbsp;0.3427 &nbsp; &nbsp;0.6086 &nbsp;-0.1384
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H11 &nbsp; &nbsp; &nbsp;37 &nbsp; &nbsp; &nbsp;1.835 &nbsp; &nbsp; &nbsp;1.181 &nbsp; &nbsp; &nbsp;3.105 &nbsp;-0.3026 &nbsp;-1.0577 &nbsp; &nbsp;1.3633
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H12 &nbsp; &nbsp; &nbsp;38 &nbsp; &nbsp; &nbsp;1.990 &nbsp; &nbsp; &nbsp;1.094 &nbsp; &nbsp; &nbsp;3.274 &nbsp; &nbsp;0.8838 &nbsp;-3.0362 &nbsp;-3.3538
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H13 &nbsp; &nbsp; &nbsp;39 &nbsp; &nbsp; &nbsp;1.767 &nbsp; &nbsp; &nbsp;1.282 &nbsp; &nbsp; &nbsp;3.603 &nbsp; &nbsp;1.5678 &nbsp; &nbsp;0.4610 &nbsp; &nbsp;0.1510
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H14 &nbsp; &nbsp; &nbsp;40 &nbsp; &nbsp; &nbsp;1.965 &nbsp; &nbsp; &nbsp;1.147 &nbsp; &nbsp; &nbsp;3.523 &nbsp; &nbsp;2.8042 &nbsp; &nbsp;0.8007 &nbsp; &nbsp;1.6468
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H15 &nbsp; &nbsp; &nbsp;41 &nbsp; &nbsp; &nbsp;1.618 &nbsp; &nbsp; &nbsp;1.301 &nbsp; &nbsp; &nbsp;2.980 &nbsp;-2.3607 &nbsp;-1.0238 &nbsp; &nbsp;1.5835
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H16 &nbsp; &nbsp; &nbsp;42 &nbsp; &nbsp; &nbsp;1.411 &nbsp; &nbsp; &nbsp;1.434 &nbsp; &nbsp; &nbsp;2.933 &nbsp;-0.4707 &nbsp;-0.3738 &nbsp;-0.3861
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H17 &nbsp; &nbsp; &nbsp;43 &nbsp; &nbsp; &nbsp;1.353 &nbsp; &nbsp; &nbsp;1.498 &nbsp; &nbsp; &nbsp;3.365 &nbsp;-0.1283 &nbsp; &nbsp;2.4331 &nbsp; &nbsp;0.5324
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H18 &nbsp; &nbsp; &nbsp;44 &nbsp; &nbsp; &nbsp;1.263 &nbsp; &nbsp; &nbsp;1.521 &nbsp; &nbsp; &nbsp;3.117 &nbsp;-2.4200 &nbsp;-1.8720 &nbsp; &nbsp;1.1648
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H19 &nbsp; &nbsp; &nbsp;45 &nbsp; &nbsp; &nbsp;0.952 &nbsp; &nbsp; &nbsp;1.545 &nbsp; &nbsp; &nbsp;3.415 &nbsp; &nbsp;0.9236 &nbsp;-0.1758 &nbsp; &nbsp;0.1509
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H1A &nbsp; &nbsp; &nbsp;46 &nbsp; &nbsp; &nbsp;1.179 &nbsp; &nbsp; &nbsp;1.495 &nbsp; &nbsp; &nbsp;3.515 &nbsp; &nbsp;0.6316 &nbsp; &nbsp;0.7709 &nbsp;-1.4005
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H1B &nbsp; &nbsp; &nbsp;47 &nbsp; &nbsp; &nbsp;0.742 &nbsp; &nbsp; &nbsp;1.462 &nbsp; &nbsp; &nbsp;3.523 &nbsp;-0.4545 &nbsp; &nbsp;0.5913 &nbsp;-0.5083
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H1C &nbsp; &nbsp; &nbsp;48 &nbsp; &nbsp; &nbsp;0.749 &nbsp; &nbsp; &nbsp;1.363 &nbsp; &nbsp; &nbsp;3.760 &nbsp; &nbsp;0.4201 &nbsp; &nbsp;1.1105 &nbsp; &nbsp;2.4743
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H1D &nbsp; &nbsp; &nbsp;49 &nbsp; &nbsp; &nbsp;1.347 &nbsp; &nbsp; &nbsp;1.280 &nbsp; &nbsp; &nbsp;3.982 &nbsp;-1.0797 &nbsp; &nbsp;0.6533 &nbsp; &nbsp;0.4476
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H1E &nbsp; &nbsp; &nbsp;50 &nbsp; &nbsp; &nbsp;1.230 &nbsp; &nbsp; &nbsp;1.205 &nbsp; &nbsp; &nbsp;0.144 &nbsp; &nbsp;0.4106 &nbsp; &nbsp;1.5828 &nbsp; &nbsp;1.9807
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H1F &nbsp; &nbsp; &nbsp;51 &nbsp; &nbsp; &nbsp;0.995 &nbsp; &nbsp; &nbsp;1.147 &nbsp; &nbsp; &nbsp;0.137 &nbsp; &nbsp;1.2287 &nbsp; &nbsp;0.4955 &nbsp; &nbsp;1.6285
 &nbsp; &nbsp; &nbsp; &nbsp;1LIG &nbsp; &nbsp; &nbsp; &nbsp;H1G &nbsp; &nbsp; &nbsp;52 &nbsp; &nbsp; &nbsp;0.847 &nbsp; &nbsp; &nbsp;1.214 &nbsp; &nbsp; &nbsp;3.994 &nbsp;-0.4029 &nbsp;-1.2299 &nbsp; &nbsp;1.5198

</details>

`type` of `atoms` must be following:

<details>
<summary>Expected result</summary>

> 'C'
'C'
'C'
'C'
'C'
'C'
'H'
'H'
'H'
'H'
'N'
'N'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'

</details>

## Actual behavior ##

However, what I actually get is:

<details>
<summary>Actual result</summary>

> 'C'
'C'
'C'
'C'
'C'
'C'
'H'
'H'
'H'
'H'
'NA'
'N'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'C'
'CS'
'C'
'CU'
'C'
'C'
'C'
'C'
'C'
'C'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'
'H'

</details>

## Code to reproduce the behavior ##

``` python
import MDAnalysis as mda
gro = mda.Universe('prod.gro')  #Example gro file
mol = gro.residues[0]

for atom in mol.atoms:
    atom.type


```

## Current version of MDAnalysis ##
- MDAnalysis : 2.5.0
- Python : 3.11.3
- operating system : Rocky Linux

### PR DESCRIPTION
**Pull request by pillose**
Fixes #4167

Changes made in this Pull Request:
 - Suppose the function `guess_atom_element()` receives 'N0A' as an argument 
 - With previous code, 
  `name = 'NA'  # interpreted as sodium`
  Which substitute '0' to ''

 - With new code, 
 `name = ['N', 'A'][0]   #== 'N"`

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

----
:books: Documentation preview :books:: https://mdanalysis--4168.org.readthedocs.build/en/4168/

### CODE REVIEW

**Review by hmacdope — CHANGES_REQUESTED** [review]
Hi @pillose! Thanks for the contribution!

Would you be able to add a test of the new behaviour? If you need help getting started with the test suite, feel free to ping me.  You will also need to add an entry to `package/CHANGELOG`.

Also as you are a new contributor (welcome!) please add yourself to `package/AUTHORS` and also introduce yourself on the [developer mailing list](https://groups.google.com/g/mdnalysis-devel) or [discord](https://discord.gg/sBXr2eJm).

**Comment by IAlibay** [comment]
@pillose this PR has been opened for a little while, are you still interested in contributing this code?

**Comment by pillose** [comment]
@hmacdope I added some test and change log. Also, I added my name(Geongi Moon) to `package/AUTHORS`.
I apologize for missing your comment.

@IAlibay Thank you for the reminder.

File: package/CHANGELOG [hunk 1]
```diff
@@ -19,6 +19,7 @@ The rules for this file:
 
 Fixes
   * Fix Boltzmann typo in `units.py` (PR #4214, Issue #4213)
+  * Fix Atom type guessing error (PR #4168, Issue #4167)
```
Comment (hmacdope) [inline package/CHANGELOG]: Order this newest first.

File: package/MDAnalysis/topology/guessers.py [hunk 2]
```diff
@@ -204,7 +204,12 @@ def guess_atom_element(atomname):
     except KeyError:
         # strip symbols and numbers
         no_symbols = re.sub(SYMBOLS, '', atomname)
-        name = re.sub(NUMBERS, '', no_symbols).upper()
+        no_numbers = re.split(NUMBERS, no_symbols)
```
Comment (hmacdope) [inline package/MDAnalysis/topology/guessers.py]: Add some comments as to why this is now in here.

**Review by hmacdope — CHANGES_REQUESTED** [review]
Looks good, two little comments. Did you end up introducing yourself on mailing list or discord :) ?

**Review by IAlibay — CHANGES_REQUESTED** [review]
@pillose - We are changing the license under which future contributions are released. Could you please confirm that you agree to releasing this code under the terms of the [LGPLv2.1](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html) and that your contribution also adheres to the [developer certificate of origion](https://developercertificate.org/)?

File: package/CHANGELOG [hunk 3]
```diff
@@ -14,11 +14,12 @@ The rules for this file:
 
 ------------------------------------------------------------------------------
 
-??/??/?? IAlibay, jaclark5, MohitKumar020291, orionarcher, xhgchen, ztimol
+??/??/?? IAlibay, jaclark5, MohitKumar020291, orionarcher, xhgchen, ztimol, pillose
```
Comment (RMeli) [inline package/CHANGELOG]:
```suggestion
??/??/?? IAlibay, jaclark5, MohitKumar020291, orionarcher, xhgchen, ztimol, 
         pillose
```

File: package/AUTHORS [hunk 4]
```diff
@@ -219,7 +219,11 @@ Chronological list of authors
   - Daniel J. Evans
   - Mohit Kumar
   - Shubham Kumar
+<<<<<<< HEAD
+  - GeongiMoon
+=======
   - Zaheer Timol
+>>>>>>> 66a2b215760750cdf5e62725249c6971e3ac8b34
```
Comment (RMeli) [inline package/AUTHORS]:
```suggestion
  - Zaheer Timol
  - GeongiMoon
```

File: package/MDAnalysis/topology/guessers.py [hunk 5]
```diff
@@ -202,9 +202,17 @@ def guess_atom_element(atomname):
...
-        name = re.sub(NUMBERS, '', no_symbols).upper()
+
+        # split name by numbers
+        no_numbers = re.split(NUMBERS, no_symbols)
+        no_numbers = list(filter(None, no_numbers)) #remove ''
+        # if no_numbers is not empty, use the first element of no_numbers
+        if no_numbers:
+            name = list(filter(None, no_numbers))[0].upper()
+        else:
+            name = ''
```
Comment (RMeli) [inline package/MDAnalysis/topology/guessers.py]:
```suggestion
        name = list(filter(None, no_numbers))[0].upper() if no_numbers else ''
```

**Review by RMeli — COMMENTED** [review]
Thanks @pillose. Please see https://github.com/MDAnalysis/mdanalysis/pull/4168#pullrequestreview-1554498651, thanks.

**Comment by pillose** [comment]
@IAlibay I agree to changing the license.
@hmacdope I just introduced myself on mailing list.
@RMeli Thank you for the reminder.

File: package/AUTHORS [hunk 6]
```diff
@@ -220,6 +220,7 @@ Chronological list of authors
   - Mohit Kumar
   - Shubham Kumar
   - Zaheer Timol
+  - GeongiMoon
```
Comment (hmacdope) [inline package/AUTHORS]: Needs a space, I am assuming?

**Review by hmacdope — CHANGES_REQUESTED** [review]

File: package/AUTHORS (hunk 6 above)
Comment (hmacdope) [reply package/AUTHORS]: If not let me know, other than that looks good.

Comment (pillose) [reply package/AUTHORS]: I missed a space... Thank you!

File: testsuite/MDAnalysisTests/topology/test_guessers.py [hunk 7]
```diff
@@ -104,6 +104,9 @@ def test_guess_atom_element_1H(self):
         ('zn', 'ZN'),
         ('Ca2+', 'CA'),
         ('CA', 'C'),
+        ('N0A', 'N'),
```
Comment (hmacdope) [inline testsuite/MDAnalysisTests/topology/test_guessers.py:107]: One more nit sorry could you test  Na+ -> Na?

**Review by hmacdope — CHANGES_REQUESTED** [review]

File: testsuite/MDAnalysisTests/topology/test_guessers.py (hunk 7 above)
Comment (pillose) [reply testsuite/MDAnalysisTests/topology/test_guessers.py:107]: I just added two tests for Na+ and Cu2+

File: package/MDAnalysis/topology/guessers.py [hunk 8]
```diff
@@ -202,9 +202,14 @@ def guess_atom_element(atomname):
...
-        # strip symbols and numbers
+        # strip symbols
         no_symbols = re.sub(SYMBOLS, '', atomname)
-        name = re.sub(NUMBERS, '', no_symbols).upper()
+
+        # split name by numbers
+        no_numbers = re.split(NUMBERS, no_symbols)
+        no_numbers = list(filter(None, no_numbers)) #remove ''
+        # if no_numbers is not empty, use the first element of no_numbers
+        name = list(filter(None, no_numbers))[0].upper() if no_numbers else ''
```
Comment (IAlibay) [inline package/MDAnalysis/topology/guessers.py]: Apologies if I'm missing something obvious, but do you need to apply filter again here? You've already removed the `''` entries and you shouldn't have any `None` entries ?

**Review by IAlibay — CHANGES_REQUESTED** [review]
The one question from me.

File: package/MDAnalysis/topology/guessers.py (hunk 8 above)
Comment (pillose) [reply package/MDAnalysis/topology/guessers.py]: Sorry, I think I made a mistake while modifying the code...

**Review by IAlibay — APPROVED** [review]
Thanks, sorry for this taking so long. lgtm!

**Review by hmacdope — APPROVED** [review]
LGTM also thanks @pillose

**Comment by hmacdope** [comment]
Cycling CI

**Pull request merged**